android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java

```diff
@@ -233,6 +239,24 @@ private boolean migrateTotalStatsAndPreferences() {
         if (!BraveUpgradeJobIntentService.needToMigratePreferences()) {
             return needToRestart;
         }
+
+        // Migrate ref code
+        SharedPreferences prefStatsFromTabs = ContextUtils.getApplicationContext()
+                .getSharedPreferences(PREF_STATS_PREFERENCES_NAME, 0);
+        String urpc = prefStatsFromTabs.getString(PREF_STATS_URPC_NAME, null);
+        if (urpc != null && !urpc.isEmpty()) {
+            String promoCodeFilePath = ContextUtils.getApplicationContext().getApplicationInfo().dataDir
+                    + File.separator + APP_CHROME_DIR + File.separator + PROMO_CODE_FILE_NAME;
+            try {
+                File promoCodeFile = new File(promoCodeFilePath);
+                FileOutputStream outputStreamWriter = new FileOutputStream(promoCodeFile);
+                outputStreamWriter.write(urpc.getBytes());
+                outputStreamWriter.close();
+            } catch (IOException e) {
+                Log.e(TAG, "Could not write to file (" + promoCodeFilePath + "): " + e.getMessage());
+            }
```

Comment:
This will migrate the referral code but I think it will also have the side effect of essentially resetting its expiration time, right? Also, would this mean that we'd be trying to initialize a referral twice (once on the old browser and then once when upgrading to the new browser?) Not sure if that's something the backend could handle.

Maybe that's OK? Not sure, just wanted to mention it though...

Reply:
good point @emerick, I added the rest of referral info for migration